toml/src/main/kotlin/org/rust/toml/crates/local/CratesLocalIndexService.kt

```diff
@@ -18,6 +20,10 @@ interface CratesLocalIndexService {
 }
 
 data class CargoRegistryCrate(val versions: List<CargoRegistryCrateVersion>) {
+    val sortedVersions: List<CargoRegistryCrateVersion> by lazy {
```

Comment:
Is it really useful to keep this list as a field? AFAIU `getCrate` invokes `read` which instantiates a new  `CargoRegistryCrate` every time. So a new `CargoRegistryCrate` object is created on every code completion session.

Related to `semanticVersion` as well

Reply:
As far as I tested, it seems that PersistentHashMap keeps the structures in the memory and doesn't reinitialize them if there is no need for that. At least the breakpoints on the constructors are being called only the first time I call completion